src/Umbraco.Core/Exceptions/WontImplementException.cs

```diff
@@ -1,27 +1,52 @@
 ﻿using System;
+using System.Runtime.Serialization;
 
 namespace Umbraco.Core.Exceptions
 {
     /// <summary>
     /// The exception that is thrown when a requested method or operation is not, and will not be, implemented.
     /// </summary>
-    /// <remarks>The <see cref="NotImplementedException"/> is to be used when some code is not implemented,
+    /// <remarks>
+    /// The <see cref="NotImplementedException" /> is to be used when some code is not implemented,
     /// but should eventually be implemented (i.e. work in progress) and is reported by tools such as ReSharper.
     /// This exception is to be used when some code is not implemented, and is not meant to be, for whatever
-    /// reason.</remarks>
+    /// reason.
+    /// </remarks>
+    /// <seealso cref="System.NotImplementedException" />
+    [Serializable]
+    [Obsolete("If a method or operation is not, and will not be, implemented, it is invalid, so we should throw an InvalidOperationException instead.")]
```

Comment:
I'm unsure about this one - if it's not implemented and doesn't intend to be, shouldn't it throw a `NotSupportedException`?

Reply:
Both `NotSupportedException` and `InvalidOperationException` are a lot better compared to the custom exception based on `NotImplementedException` and deciding which one to throw probably depends on the context, see:
https://docs.microsoft.com/en-us/dotnet/api/system.notsupportedexception and https://docs.microsoft.com/en-us/dotnet/api/system.invalidoperationexception.